lib/licensed/sources/yarn.rb

```diff
@@ -26,22 +26,30 @@ def enumerate_dependencies
         end
       end
 
+      # Finds packages that the current project relies on
       def packages
-        root_dependencies = JSON.parse(yarn_list_command)["data"]["trees"]
-        root_path = config.pwd
+        return [] if yarn_package_tree.nil?
         all_dependencies = {}
-        recursive_dependencies(root_path, root_dependencies).each do |name, results|
+        recursive_dependencies(config.pwd, yarn_package_tree).each do |name, results|
           results.uniq! { |package| package["version"] }
           if results.size == 1
+            # if there is only one package for a name, reference it by name
             all_dependencies[name] = results[0]
           else
+            # if there is more than one package for a name, reference each by
+            # "<name>-<version>"
             results.each do |package|
               all_dependencies[package["id"].sub("@", "-")] = package
```

Comment:
I'm not entirely sure about this, for example, consider package named `@apollo/react-common`, with the version it becomes `@apollo/react-common@3.1.3`, so this line would cause the first `@` to be replaced too. Maybe something like `.sub(/(.)@(.)/, '\1-\2')` would work?

Especially that `npm` source does not strip the leading `@`. Check this branch: https://github.com/krzysztof-pawlik-gat/licensed/tree/ensure-at-in-packages (or compare view: https://github.com/github/licensed/compare/master...krzysztof-pawlik-gat:ensure-at-in-packages?expand=1).

Reply:
🙏  great catch, I forgot about scoped packages.  updated in https://github.com/github/licensed/pull/233/commits/9005eda88cbfc0b6c9460ea508fe2af2436ed3db